dipy/workflows/io.py

```diff
@@ -324,6 +332,259 @@ def run(self, input_files, vol_idx=0, out_dir="", out_split="split.nii.gz"):
+            logging.info("Extracting b0 from {0}".format(dwi))
+            data, affine, image = load_nifti(dwi, return_img=True)
+
+            bvals, bvecs = read_bvals_bvecs(bval, bvec)
+            # If all b-values are smaller or equal to the b0 threshold, it is
+            # assumed that no thresholding is requested
+            if any(mask_non_weighted_bvals(bvals, b0_threshold)):
+                if b0_threshold < bvals.min():
+                    warnings.warn(
+                        f"b0_threshold (value: {b0_threshold}) is too low, "
+                        "increase your b0_threshold. It should be higher than the "
+                        f"first b0 value ({bvals.min()}).",
+                        stacklevel=2,
+                    )
+            gtab = gradient_table(
+                bvals, bvecs=bvecs, b0_threshold=b0_threshold, atol=bvecs_tol
+            )
+            b0s_result = extract_b0(
+                data,
+                gtab.b0s_mask,
+                group_contiguous_b0=group_contiguous_b0,
+                strategy=strategy,
+                chunk_size=chunk_size,
+            )
+
+            if b0s_result.ndim == 3:
+                save_nifti(ob0, b0s_result, affine, hdr=image.header)
+                logging.info("b0 saved as {0}".format(ob0))
+            elif b0s_result.ndim == 4:
+                for i in range(b0s_result.shape[-1]):
+                    save_nifti(
+                        ob0.replace(".nii", f"_{i}.nii"),
+                        b0s_result[..., i],
+                        affine,
+                        hdr=image.header,
+                    )
+                    logging.info(
+                        "b0 saved as {0}".format(ob0.replace(".nii", f"_{i}.nii"))
+                    )
+            else:
+                logging.error("No b0 volumes found")
+
+
+class ExtractShellFlow(Workflow):
+    @classmethod
+    def get_short_name(cls):
+        return "extract_shell"
+
+    def run(
+        self,
+        input_files,
+        bvalues_files,
+        bvectors_files,
+        bvals_to_extract=None,
+        b0_threshold=50,
+        bvecs_tol=0.01,
+        tol=20,
+        chunk_size=None,
+        group_shells=True,
+        out_dir="",
+        out_shell="shell.nii.gz",
+    ):
+        """Extract shells from the input 4D file.
+
+        Parameters
+        ----------
+        input_files : string
+            Path to the input volumes. This path may contain wildcards to
+            process multiple inputs at once.
+        bvalues_files : string
+            Path to the bvalues files. This path may contain wildcards to use
+            multiple bvalues files at once.
+        bvectors_files : string
+            Path to the bvectors files. This path may contain wildcards to use
+            multiple bvectors files at once.
+        bvals_to_extract : variable int, optional
+            List of b-values to extract.
+        b0_threshold : float, optional
```

Comment:
Some of the parameters here, b0_threshold, bvecs_tol seem to change the behavior a lot from the original function. If it is an important behavior shouldn't it be in the actual function not just on the workflow?

Reply:
since the original function expect a `gtab` (gradient table), `b0_threshold` and `bvecs_tol` are implicitly needed when the user will create his gradient table.  The original function assumes that this work has been done before